analysis/config/config.go

```diff
@@ -222,26 +208,47 @@ type Options struct {
 	// ReportNoCalleeSites specifies whether the tool should report where it does not find any callee.
 	ReportNoCalleeSites bool `xml:"report-no-callee-sites,attr" yaml:"report-no-callee-sites" json:"report-no-callee-sites"`
 
-	// MaxDepth sets a limit for the number of function call depth explored during the analysis
-	// Default is -1.
-	// If provided MaxDepth is <= 0, then it is ignored.
-	MaxDepth int `xml:"max-depth,attr" yaml:"max-depth" json:"max-depth"`
+	// ReportPaths specifies whether the taint flows should be reported in separate files. For each taint flow, a new
+	// file named taint-*.out will be generated with the trace from source to sink
+	ReportPaths bool `xml:"report-paths,attr" yaml:"report-paths" json:"report-paths"`
 
-	// PathSensitive is a boolean indicating whether the analysis should be run with access path sensitivity on
-	// (will change to include more filtering in the future)
-	// Note that the configuration option name is "field-sensitive" because this is the name that will be more
-	// recognizable for users.
-	PathSensitive bool `xml:"field-sensitive" yaml:"field-sensitive" json:"field-sensitive"`
+	// ReportSummaries can be set to true, in which case summaries will be reported in a file names summaries-*.out in
+	// the reports directory
+	ReportSummaries bool `xml:"report-summaries,attr" yaml:"report-summaries" json:"report-summaries"`
 
-	// MaxAlarms sets a limit for the number of alarms reported by an analysis.  If MaxAlarms > 0, then at most
-	// MaxAlarms will be reported. Otherwise, if MaxAlarms <= 0, it is ignored.
-	MaxAlarms int `xml:"max-alarms,attr" yaml:"max-alarms" json:"max-alarms"`
+	// ReportsDir is the directory where all the reports will be stored. If the yaml config file this config struct has
+	// been loaded does not specify a ReportsDir but sets any Report* option to true, then ReportsDir will be created
+	// in the folder the binary is called.
+	ReportsDir string `xml:"reports-dir,attr" yaml:"reports-dir" json:"reports-dir"`
 
-	// Loglevel controls the verbosity of the tool
-	LogLevel int `xml:"log-level,attr" yaml:"log-level" json:"log-level"`
+	// SkipInterprocedural can be set to true to skip the interprocedural (inter-procedural analysis) step
+	SkipInterprocedural bool `xml:"skip-interprocedural,attr" yaml:"skip-interprocedural" json:"skip-interprocedural"`
 
 	// Suppress warnings
 	SilenceWarn bool `xml:"silence-warn,attr" json:"silence-warn" yaml:"silence-warn"`
+
+	// SourceTaintsArgs specifies whether calls to a source function also taints the argument. This is usually not
+	// the case, but might be useful for some users or for source functions that do not return anything.
+	SourceTaintsArgs bool `xml:"source-taints-args,attr" yaml:"source-taints-args" json:"source-taints-args"`
```

Comment:
We can leave this for a future PR but I think analysis-specific options should be part of that specific analysis' configuration, not the global options set

Reply:
Yes, I think once the options settle we should do some reorganization.